mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/VectorField.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.definition.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.Searchable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+import org.hibernate.search.engine.environment.bean.BeanRetrieval;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMapping;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessorRef;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.impl.VectorFieldAnnotationProcessor;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * Maps an entity property to a vector field in the index.
+ * <p>
+ * This annotation will work for any {@code float/byte} array.
+ * <p>
+ * Vector fields are to be used in k-NN search, when the distance between a queried and stored vectors is computed
+ * and k nearest vectors are selected as the results.
+ *
+ * @see org.hibernate.search.engine.search.predicate.dsl.KnnPredicateFieldStep
+ */
+@Documented
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(VectorField.List.class)
+@PropertyMapping(processor = @PropertyMappingAnnotationProcessorRef(type = VectorFieldAnnotationProcessor.class,
+		retrieval = BeanRetrieval.CONSTRUCTOR))
+@Incubating
+public @interface VectorField {
```

Comment:
I was looking into problems one might encounter because of the mandatory number of dimensions, and opened https://hibernate.atlassian.net/browse/HSEARCH-5021, but I just noticed... this annotation doesn't support passing a bridge/binder at all?

Feel free to repurpose https://hibernate.atlassian.net/browse/HSEARCH-5021 to address this later if necessary.